plugins/router/arris-tg3442

```diff
@@ -0,0 +1,250 @@
+    )
+
+    # set session
+    r = session.post(f"{url}/php/ajaxSet_Session.php")
+
+def docsis_status(session):
+    """get current DOCSIS status page, parse and return channel data"""
+    r = session.get(f"{url}/php/status_docsis_data.php")
+    # extract json from javascript
+    json_downstream_data = re.search(r".*json_dsData = (.+);.*", r.text)[1]
+    json_upstream_data = re.search(r".*json_usData = (.+);.*", r.text)[1]
+    # parse json
+    downstream_data = json.loads(json_downstream_data)
+    upstream_data = json.loads(json_upstream_data)
+    # convert lock status to numeric values
+    for d in [ upstream_data, downstream_data ]:
+        for c in d:
+            if c['LockStatus'] == "ACTIVE" or c['LockStatus'] == "Locked":
+                c['LockStatus'] = 1
+            else:
+                c['LockStatus'] = 0
+    return downstream_data, upstream_data
+
+
+# -----------------------------------------------------------------------------
+if __name__ == "__main__":
+    # get config
+    url = os.getenv("url")
+    username = os.getenv("username")
+    password = os.getenv("password")
+    # validate config
+    if not url or not username or not password:
+        print("Set url, username and password first.", file=sys.stderr)
+        exit(1)
+    # create session
+    session = requests.Session()
+    # login with username and password
+    login(session, url, username, password)
+    # get DOCSIS status
+    downstream, upstream = docsis_status(session)
+    # prepare munin graph info
+    graph_descriptions = [
+        {
+            "name": "up_signal",
+            "title": "DOCSIS Upstream signal strength",
+            "vlabel": "dBmV",
+            "info": "DOCSIS upstream signal strength by channel",
+            "data": upstream,
+            "key": "PowerLevel"
+        },
+        {
+            "name": "up_lock",
+            "title": "DOCSIS Upstream lock",
+            "vlabel": "locked",
+            "info": "DOCSIS upstream channel lock status",
+            "data": upstream,
+            "key": "LockStatus"
+        },
+        {
+            "name": "down_signal",
+            "title": "DOCSIS Downstream signal strength",
+            "vlabel": "dBmV",
+            "info": "DOCSIS downstream signal strength by channel",
+            "data": downstream,
+            "key": "PowerLevel"
+        },
+        {
+            "name": "down_lock",
+            "title": "DOCSIS Downstream lock",
+            "vlabel": "locked",
+            "info": "DOCSIS downstream channel lock status",
+            "data": downstream,
+            "key": "LockStatus"
+        },
+        {
+            "name": "down_snr",
+            "title": "DOCSIS Downstream signal/noise ratio",
+            "vlabel": "dB",
+            "info": "SNR/MER",
+            "data": downstream,
+            "key": "SNRLevel"
+        }
+    ]
+
+    # configure ?
+    if len(sys.argv) > 1 and "config" in sys.argv[1]:
```

Comment:
Here `config == sys.argv[1]` is probably, what you intended?
(instead of `in`)

Reply:
It was intended but I admit, it might be a bit too defensive.